model/host/host.go

```diff
@@ -1600,6 +1600,37 @@ func (h *Host) AddSSHKeyName(name string) error {
 	return nil
 }
 
+// HandleTerminatedHostSpawnedByTask should be called when a host spawned by a task
```

Comment:
I would rephrase this doc to instead say something along the lines of "re-creates a new intent host if possible when this host.create host fails; otherwise if it cannot create a new host, it'll populate the reason that host.create failed".

Reply:
done (though now it's private)